File: backend/app/models/feedback.py
"""Elasticsearch index and constants for feedback items."""

FEEDBACK_SOURCES = [
    "app_store_review",
    "g2_capterra",
    "support_ticket",
    "nps_csat",
    "customer_email",
    "sales_call_note",
    "slack_message",
    "internal_team_feedback",
    "user_interview",
    "bug_report",
    "community_forum",
]

FEEDBACK_SOURCE_LABELS = {
    "app_store_review": "App Store Review",
    "g2_capterra": "G2 / Capterra",
    "support_ticket": "Support Ticket",
    "nps_csat": "NPS / CSAT Survey",
    "customer_email": "Customer Email",
    "sales_call_note": "Sales Call Note",
    "slack_message": "Slack Message",
    "internal_team_feedback": "Internal Team Feedback",
    "user_interview": "User Interview / Research",
    "bug_report": "Bug Report (Jira/Linear)",
    "community_forum": "Community Forum / Discord",
}


def feedback_index(org_id: str) -> str:
    """Return the feedback index name for an org."""
    return f"{org_id}-feedback"


FEEDBACK_MAPPING = {
    "mappings": {
        "properties": {
            "id": {"type": "keyword"},
            "org_id": {"type": "keyword"},
            "text": {"type": "text", "fields": {"keyword": {"type": "keyword"}}},
            "source": {"type": "keyword"},
            "sentiment": {"type": "keyword"},
            "sentiment_score": {"type": "float"},
            "rating": {"type": "integer"},
            "product_area": {"type": "keyword"},
            "customer_id": {"type": "keyword"},
            "customer_name": {"type": "keyword"},
            "customer_segment": {"type": "keyword"},
            "author_name": {"type": "keyword"},
            "author_email": {"type": "keyword"},
            "tags": {"type": "keyword"},
            "source_file": {"type": "keyword"},
            "ingestion_method": {"type": "keyword"},
            "created_at": {"type": "date"},
            "ingested_at": {"type": "date"},
            "metadata": {"type": "object", "enabled": True},
        }
    }
}

FEEDBACK_MAPPING_WITH_ELSER = {
    "mappings": {
        "properties": {
            "id": {"type": "keyword"},
            "org_id": {"type": "keyword"},
            "text": {"type": "text", "fields": {"keyword": {"type": "keyword"}}},
            "text_semantic": {
                "type": "semantic_text",
                "inference_id": "elser-endpoint",
            },
            "source": {"type": "keyword"},
            "sentiment": {"type": "keyword"},
            "sentiment_score": {"type": "float"},
            "rating": {"type": "integer"},
            "product_area": {"type": "keyword"},
            "customer_id": {"type": "keyword"},
            "customer_name": {"type": "keyword"},
            "customer_segment": {"type": "keyword"},
            "author_name": {"type": "keyword"},
            "author_email": {"type": "keyword"},
            "tags": {"type": "keyword"},
            "source_file": {"type": "keyword"},
            "ingestion_method": {"type": "keyword"},
            "created_at": {"type": "date"},
            "ingested_at": {"type": "date"},
            "metadata": {"type": "object", "enabled": True},
        }
    }
}

File: frontend/src/types/feedback.ts
/** Feedback item from API. */
export interface Feedback {
  id: string;
  org_id?: string;
  text: string;
  source?: string;
  sentiment?: string;
  sentiment_score?: number;
  rating?: number;
  product_area?: string;
  customer_id?: string;
  customer_name?: string;
  customer_segment?: string;
  author_name?: string;
  author_email?: string;
  tags?: string[];
  source_file?: string;
  ingestion_method?: string;
  created_at?: string;
  ingested_at?: string;
  metadata?: Record<string, unknown>;
}

/** Feedback source types for display (id, label, color). */
export const FEEDBACK_SOURCES: { id: string; label: string; color: string }[] = [
  { id: "app_store_review", label: "App Store Review", color: "blue" },
  { id: "g2_capterra", label: "G2 / Capterra", color: "indigo" },
  { id: "support_ticket", label: "Support Ticket", color: "orange" },
  { id: "nps_csat", label: "NPS / CSAT Survey", color: "green" },
  { id: "customer_email", label: "Customer Email", color: "teal" },
  { id: "sales_call_note", label: "Sales Call Note", color: "purple" },
  { id: "slack_message", label: "Slack Message", color: "pink" },
  { id: "internal_team_feedback", label: "Internal Team Feedback", color: "gray" },
  { id: "user_interview", label: "User Interview / Research", color: "cyan" },
  { id: "bug_report", label: "Bug Report (Jira/Linear)", color: "red" },
  { id: "community_forum", label: "Community Forum / Discord", color: "yellow" },
];

/** CSV upload init response. */
export interface FeedbackUploadInit {
  upload_id: string;
  columns: string[];
  suggested_mapping: Record<string, string | null>;
  total_rows: number;
  preview_sample?: Record<string, string>[];
}

/** CSV confirm request. */
export interface FeedbackUploadConfirm {
  column_mapping: Record<string, string | null>;
  default_source?: string;
  use_today_for_date?: boolean;
  auto_detect_areas?: boolean;
  auto_analyze_sentiment?: boolean;
}

/** Detected area from import. */
export interface DetectedArea {
  name: string;
  count: number;
  is_new: boolean;
}

/** CSV import result. */
export interface FeedbackUploadImport {
  upload_id: string;
  total_rows: number;
  imported_rows: number;
  failed_rows: number;
  detected_areas?: DetectedArea[];
}
